### ISSUE TITLE
Product Attribute Repository do not respect attribute_code max length

### ISSUE THREAD

**Issue by drc0** [magento/magento2#29017]
### Preconditions (*)
on magento 2.4 a product attribute code with attribute_code of value A_12345678_9123_4567_890a_BCDEF123456 is saved through the admin ui but not from the repository. Here Magento\Eav\Model\Entity\Attribute::ATTRIBUTE_CODE_MAX_LENGTH should be respected.

### Steps to reproduce (*)

Here is a code example:
```
$objectManager = \Magento\Framework\App\ObjectManager::getInstance();
$ar = $objectManager->create('Magento\Catalog\Api\ProductAttributeRepositoryInterface');
$attf = $objectManager->create('\Magento\Catalog\Api\Data\ProductAttributeInterfaceFactory');
$a = $attf->create();
$a->setData(['attribute_code' => 'A_12345678_9123_4567_890a_BCDEF123456','frontend_label' => 'a test', 'frontend_input' => 'text']);
$ar->save($a);
    Magento/Framework/Exception/InputException with message 'Invalid value of "A_12345678_9123_4567_890a_BCDEF123456" provided for the attribute_code field.'


```
### Expected result (*)

1. attribute is saved

### Actual result (*)

1. validator exception 

### Addiotional information
Issue is reproducible via WebApi
![#29017Main](https://user-images.githubusercontent.com/51681379/94140580-1af2c280-fe74-11ea-8ee7-bf94240c079e.png)

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Charlie** [comment]
Hi, @drc0 I think that issue is in absent validation in admin.
What do you

**Comment by drc0** [comment]
>  I think that issue is in absent attribute code validation via admin.

with magento 2.4.0 I'm not able to reproduce the admin issue anymore (the string is 61 characters ), do you?
![Screenshot_20200902_193010](https://user-images.githubusercontent.com/5507633/92016774-1cd4d480-ed53-11ea-9431-4c96880edc1f.png)

**Comment by engcom-Charlie** [comment]
@drc0 so can I close the issue?

**Comment by drc0** [comment]
Sure. Thanks. 

Il 3 settembre 2020 08:55:50 CEST, Vadim Malesh <notifications@github.com> ha scritto:
>@drc0 so can I close the issue?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/magento/magento2/issues/29017#issuecomment-686294566

-- 
Inviato dal mio dispositivo Android con K-9 Mail. Perdonate la brevità.

**Comment by sidolov** [comment]
The issue should be reproduced via WebApi call during the new attribute creation since the logic is absent in the repoistory

**Comment by drc0** [comment]
@sidolov sorry I'm not sure I understood your comment, but a POST to `/V1/products/attributes` with this payload and the patch applied we can verify that the check is respected (and not hardcoded to 30 chars):
here the attribute is created:

```
{
  "attribute": {
    "extension_attributes": {},
    "is_wysiwyg_enabled": true,
    "is_html_allowed_on_front": true,
    "used_for_sort_by": true,
    "is_filterable": true,
    "is_filterable_in_search": true,
    "is_used_in_grid": true,
    "is_visible_in_grid": true,
    "is_filterable_in_grid": true,
    "position": 0,
    "apply_to": [
      "string"
    ],
    "is_searchable": "string",
    "is_visible_in_advanced_search": "string",
    "is_comparable": "string",
    "is_used_for_promo_rules": "string",
    "is_visible_on_front": "string",
    "used_in_product_listing": "string",
    "is_visible": true,
    "scope": "string",
    "attribute_id": 0,
    "attribute_code":  "A_12345678_9123_4567_890a_BCDEF12345612312312312312312312312",
    "frontend_input": "text",

    "is_required": true,
    
    "is_user_defined": true,
    "default_frontend_label": "string",
    "frontend_labels": [
      {
        "store_id": 0,
        "label": "string"
      }
    ]
    
  }
}
```

while a POST with an attribute_code of 61 chars (\Magento\Eav\Model\Entity\Attribute::ATTRIBUTE_CODE_MAX_LENGTH is 60), fails with this error `abstractdb An attribute code must not be less than 1 and more than 60 characters.{"message":"An attribute code must not be less than 1 and more than 60 characters.","trace":"#0 \/var\/www\/project\/vendor\/magento\/module-catalog\/Model\/ResourceModel\/Eav\/Attribute.php(228): ......`

**Comment by magento-engcom-team** [comment]
@engcom-Delta Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Component: "` label(s) to this ticket based on verification result. If uncertain, you may follow the best guess

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-37916` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @drc0. Thank you for your report. 
The issue has been fixed in magento/magento2#29015 by @drc0 in 2.4-develop branch
Related commit(s):
 - [d2c352d0fff685cb786441a27495c8dc5be601aa](https://github.com/magento/magento2/commit/d2c352d0fff685cb786441a27495c8dc5be601aa)
 - [37dbcf76f8313c87ad8454c539758927f3943459](https://github.com/magento/magento2/commit/37dbcf76f8313c87ad8454c539758927f3943459)
 - [2663ff4fe52ab58eed5793feb0944b084af79d73](https://github.com/magento/magento2/commit/2663ff4fe52ab58eed5793feb0944b084af79d73)
 - [78ed8624f5125f16947e1f59731eb58be1ff0026](https://github.com/magento/magento2/commit/78ed8624f5125f16947e1f59731eb58be1ff0026)
 - [65a67aa7b647fb1a0604f31b4fb8af39ec732237](https://github.com/magento/magento2/commit/65a67aa7b647fb1a0604f31b4fb8af39ec732237)
 - [428e3d8b4730593472cc58cf91786d6f6f609306](https://github.com/magento/magento2/commit/428e3d8b4730593472cc58cf91786d6f6f609306)
 - [431bfe5f848c218db7f9a96aba3cf2b6a461e351](https://github.com/magento/magento2/commit/431bfe5f848c218db7f9a96aba3cf2b6a461e351)

The fix will be available with the upcoming 2.4.2 release.
